WordPress/src/main/java/org/wordpress/android/ui/uploads/UploadService.java

```diff
@@ -56,6 +56,15 @@
     @Inject PostStore mPostStore;
     @Inject SiteStore mSiteStore;
 
+    private class UploadingPost {
+        PostModel postModel;
+        List<MediaModel> pendingMedia = new ArrayList<>();
+        UploadingPost(PostModel postModel, List<MediaModel> pendingMedia) {
+            this.postModel = postModel;
+            this.pendingMedia = pendingMedia;
+        }
+    }
+
```

Comment:
We could - that was my original approach (with a `PostModel` --> `List<MediaModel>` map, though I also played with using post ID keys). But my take was that `UploadingPost` was more readable than another map in `UploadService`, and is also extendable when the time comes to track something else for each uploading post.

Also, I'd rather keep the `PostModel` structure wherever possible, and a `ConcurrentHashMap<PostModel, List<MediaModel>>` gets to be a little unwieldly in my opinion.

What do you think the advantage is to using a map instead?